tests/PHPStan/Analyser/data/array-combine.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace ArrayCombine;
+
+use function PHPStan\Testing\assertType;
+
+class Foo
+{
+	/** @phpstan-return 'foo' */
+	public function __toString(): string
+	{
+		return 'foo';
+	}
+}
+
+class Bar
+{
+	public function __toString(): string
+	{
+		return 'bar';
+	}
+}
+
+class Baz {}
+
+function withBoolKey(): void
+{
+	$a = [true, 'red', 'yellow'];
+	$b = ['avocado', 'apple', 'banana'];
+
+	assertType("array{1: 'avocado', red: 'apple', yellow: 'banana'}", array_combine($a, $b));
+
+	$c = [false, 'red', 'yellow'];
+	$d = ['avocado', 'apple', 'banana'];
+
+	assertType("array{: 'avocado', red: 'apple', yellow: 'banana'}", array_combine($c, $d));
+}
+
+function withFloatKey(): void
+{
+	$a = [1.5, 'red', 'yellow'];
+	$b = ['avocado', 'apple', 'banana'];
+
+	assertType("array{1.5: 'avocado', red: 'apple', yellow: 'banana'}", array_combine($a, $b));
+}
+
+function withIntegerKey(): void
+{
+	$a = [1, 2, 3];
+	$b = ['avocado', 'apple', 'banana'];
+
+	assertType("array{1: 'avocado', 2: 'apple', 3: 'banana'}", array_combine($a, $b));
+}
+
+function withNumericStringKey(): void
+{
+	$a = ["1", "2", "3"];
+	$b = ['avocado', 'apple', 'banana'];
+
+	assertType("array{1: 'avocado', 2: 'apple', 3: 'banana'}", array_combine($a, $b));
+}
+
+function withObjectKey() : void
+{
+	$a = [new Foo, 'red', 'yellow'];
+	$b = ['avocado', 'apple', 'banana'];
+
+	assertType("array{foo: 'avocado', red: 'apple', yellow: 'banana'}", array_combine($a, $b));
+	assertType("non-empty-array<string, 'apple'|'avocado'|'banana'>", array_combine([new Bar, 'red', 'yellow'], $b));
+	//assertType("non-empty-array<int|string, 'apple'|'avocado'|'banana'>", array_combine([new Baz, 'red', 'yellow'], $b));
```

Comment:
I'm not sure about this case. Currently `non-empty-array<'red'|'yellow'|ArrayCombine\Baz, 'apple'|'avocado'|'banana'>` is inferred. But that's not correct. `ArrayCombine\Baz` can't be converted to array key. So here should be an error type or `array-key` or something?

Reply:
since this scenario turns into a fatal error - see https://3v4l.org/puRuK - the result should be `*NEVER*`, so the `NeverType` should be returned